### PR TITLE
Change the literal delimiter to the vertical pipe character.

### LINKED ISSUE #263: Pick a delimiter for literals other than the double quote
@markusicu wrote in https://github.com/unicode-org/message-format-wg/pull/230#issuecomment-1116903103:

> Speaking of literals, don't enclose them in quotes. One of the stated goals here is to make messages usable as string literals in programming languages. Enclose literals in parentheses `{... {(5) :number} books}` or angle brackets `{... {<5> :number} books}` or pairs of pipe symbols `{... {|5| :number} books}` or similar.

### PR DESCRIPTION
**Pull request by stasm**
Closes #263.

### CODE REVIEW

File: spec/syntax.md [hunk 1]
```diff
@@ -370,19 +370,15 @@ option = name [s] "=" [s] (literal / nmtoken / variable)
...
+|1.23| :number maxFractionDigits=1
 ```
 
 ```
-(1970-01-01T13:37:00.000Z) :datetime weekday=long
-```
-
-```
-(Thu Jan 01 1970 14:37:00 GMT+0100 \(CET\)) :datetime weekday=long
+|Thu Jan 01 1970 14:37:00 GMT+0100 (CET)| :datetime weekday=long
```
Comment (aphillips) [inline spec/syntax.md:381]: Ew. Keep the ISO8601 format for the literal.

**Review by aphillips — CHANGES_REQUESTED** [review]
Mostly good, except for the modification of the date string.

File: spec/syntax.md (hunk 1 above)
Comment (stasm) [reply spec/syntax.md:381]: Keep both or keep iso8601 only?

Comment (aphillips) [reply spec/syntax.md:381]: I would only use the ISO8601 format. It doesn't make sense to have the RFC format and associate it with a formatter.

If we wanted to have a timezone, the emerging extension to 8601 for Temporals is to include the IANA/Olson name, either in the form of a zoned offset date time:

>2007-01-01T01:00:00.000+01:00[America/Chicago]

Or a zoned instant:

> 2007-01-01T01:00:00.000Z[America/Chicago]

See [here](https://w3c.github.io/timezone/#serializations) for more examples.

Comment (gibson042) [reply spec/syntax.md:381]: Agreed.
```suggestion
|1970-01-01T14:37:00+01:00[Europe/Berlin]| :datetime weekday=long
```

**Review by aphillips — APPROVED** [review]

**Review by eemeli — APPROVED** [review]

File: spec/syntax.md [hunk 2]
```diff
@@ -502,7 +498,7 @@ They are allowed in translatable text as well as in literals.
 
 ```abnf
 text-escape = "\\" / "\{" / "\}"
-literal-escape = "\\" / "\(" / "\)"
```
Comment (alerque) [inline spec/syntax.md:505]: These can probably benefit from the same refactor using a named backslash and grouping the backslash outside the set of escaped characters that @gibson042 [suggested](https://github.com/unicode-org/message-format-wg/pull/347#discussion_r1120574252) in the main ABNF PR.

**Review by alerque — APPROVED** [review]

File: spec/syntax.md [hunk 3]
```diff
@@ -502,7 +498,7 @@ They are allowed in translatable text as well as in literals.
...
+literal-escape = "\\" / "\|"
```
Comment (alerque) [inline spec/syntax.md:501]: These can probably benefit from the same refactor using a named backslash and grouping the backslash outside the set of escaped characters that @gibson042 [suggested](https://github.com/unicode-org/message-format-wg/pull/347#discussion_r1120574252) in the main ABNF PR.

**Review by alerque — APPROVED** [review]

File: spec/syntax.md (hunk 3 above)
Comment (stasm) [reply spec/syntax.md:501]: Good point. I'll rebase on top of #347 once it lands.

Comment (alerque) [reply spec/syntax.md:501]: Maybe just needs a rebase or merge from the base branch... just the literal escape bit will need a merge conflict resolution.

**Comment by stasm** [comment]
I don't seem to be able to merge this at this point due to the `license/cla` check not having finished. I don't see a way to restart it.

**Pull request merged**